src/Resources/RelationManager/EditRecord.php

```diff
@@ -52,6 +52,8 @@ public function mount()
 
     public function render()
     {
+        // TODO: figure out how to conditionally fire this for HasOne and BelongsTo
+        $this->switchRecordInstance($this->getRelationship()->first());
```

Comment:
@danharrin - note the TODO, as the manager property in this class isn't an instance we cannot use:

```php
if (
    $this->isType(Relations\HasOne::class) ||
    $this->isType(Relations\BelongsTo::class)
)
```

As I would have planned.